src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -596,33 +598,81 @@ void TriggeredPublisher::Configure(const Entity &,
     }
   }
   else
-  {
     ignerr << "No ouptut specified" << std::endl;
-    return;
+
+  if (sdfClone->HasElement("service"))
+  {
+    for (auto serviceElem = sdfClone->GetElement("service"); serviceElem;
+         serviceElem = serviceElem->GetNextElement("service"))
+    {
+       ServiceOutputInfo serviceInfo;
+       serviceInfo.servName = serviceElem->Get<std::string>("name");
+       if (serviceInfo.servName.empty())
+       {
+         ignerr << "Service name cannot be empty\n";
+       }
+       serviceInfo.reqType = serviceElem->Get<std::string>("reqType");
+       if (serviceInfo.reqType.empty())
+       {
+         ignerr << "Service request type cannot be empty\n";
+       }
+       serviceInfo.repType = serviceElem->Get<std::string>("repType");
+       if (serviceInfo.repType.empty())
+       {
+         ignerr << "Service response type cannot be empty\n";
+       }
+       serviceInfo.reqMsg = serviceElem->Get<std::string>("reqMsg");
+       if (serviceInfo.reqMsg.empty())
+       {
+         ignerr << "Service request string cannot be empty\n";
+       }
+       std::string timeoutInfo = serviceElem->Get<std::string>("timeout");
+       if (timeoutInfo.empty())
+         {
+           ignwarn << "Timeout value not specified for service name ["
+                   << serviceInfo.servName << "] with service type ["
+                   << serviceInfo.reqType << "]. Using default value of 3000\n";
+           // Use default timeout value of 3000ms
+           serviceInfo.timeout = 3000;
+         }
+       else
+           serviceInfo.timeout = std::stoi(timeoutInfo);
+
+       this->serviceOutputInfo.push_back(std::move(serviceInfo));
+    }
   }
+  else
+    ignwarn << "No service specified" << std::endl;
 
   auto msgCb = std::function<void(const transport::ProtoMsg &)>(
       [this](const auto &_msg)
       {
         if (this->MatchInput(_msg))
         {
+          if (this->delay > 0ms)
+          {
+            std::lock_guard<std::mutex> lock(this->publishQueueMutex);
+            this->publishQueue.push_back(this->delay);
+          }
+          else
           {
```

Comment:
Does removing this extra scope change the original behavior?

Reply:
are you referring to the scope line 653 to 656? I need the scope for the lock guard

Reply:
I'm referring to the outer scope in the original code that was around the entire inner if-else. That seems to be moved around in the new code? Is that layer still needed?

Reply:
Oh yeah. I took it out cause I felt it was unnecessary and a duplicate. Three was no change in behavior